NDTensors/src/dense/dense.jl

```diff
@@ -20,38 +20,43 @@ struct Dense{ElT,VecT<:AbstractArray} <: TensorStorage{ElT}
   end
 end
 
-function Dense(data::VecT) where {VecT<:AbstractArray{ElT}} where {ElT}
-  return Dense{ElT,VecT}(data)
+## We need a general typeinfo constructor for Dense
+default_Densetype(datatype::Type{<:AbstractArray{ElT}}) where {ElT} = default_storagetype(datatype)
+default_Densetype(eltype::Type{<:Number}) = default_Densetype(default_datatype(eltype))
+default_Densetype() = default_Densetype(default_datatype(default_eltype()))
+## End typeinfo based constructors 
+
+#Start with high information constructors and move to low information constructors
+function Dense{ElT, VecT}() where {ElT, VecT<:AbstractArray{ElT}}
+  return default_Densetype(VecT)(VecT())
 end
 
-function Dense(data::Array{ElT}) where {ElT}
-  return Dense{ElT,Vector{ElT}}(vec(data))
+# Construct from a set of indices
+function Dense{ElT,VecT}(inds::Tuple) where {ElT,VecT<:AbstractArray{ElT}}
+  return default_Densetype(VecT)(VecT(undef, dim(inds)))
```

Comment:
The function `default_Densetype` doesn't seem necessary here since the element type and data type are already specified by `ElT` and `VecT`, it seems like you can just call the inner constructor directly here.